src/Arcus.Security.Core/Exceptions/SecretNotFoundException.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Arcus.Security.Core.Exceptions
+{
+    /// <summary>
+    /// Exception, thrown when no secret was found, using the given name.
+    /// </summary>
+    public class SecretNotFoundException : Exception
+    {
+        public string Name { get;  }
+        public SecretNotFoundException(string name) : base($"The secret {name} was not found.")
```

Comment:
Could you assert for non-blank `string` values and maybe add another ctor to pass along the inner exception?
Thanks

Reply:
did it by leveraging Guard.Net (which is our recommended nuget package for validation) - also added innerexception & unit test for this

Reply:
:+1: